contents/go/Go.md

```diff
@@ -0,0 +1,143 @@
+
+Go is an [open source project](https://github.com/golang/go). Its source code may be useful reading for those interested in learning good practices, or simply to find out more about how the language was implemented.
+
+## Getting started
+
+Go provides its own [installation guide](https://golang.org/doc/install) and an interactive [ tour of Go](https://tour.golang.org/). These are useful and highly comprehensive resources for programmers looking to learn the syntax and style of Go. For those who prefer to read existing code examples, [Go by Example](https://gobyexample.com/) is a collection of code samples covering a wide variety of features in Go, and includes line-by-line explanations of the code. For those looking for a quick crash course on Go syntax, the Learn X In Y Minutes [Go cheatsheet](https://learnxinyminutes.com/docs/go/) may also be a good starting point.
+
+If you're unconvinced about Go, [the Go playground](https://play.golang.org/) can be used as a sandbox in which to write, build and execute code without installing Go on your machine.
+
+## Using Go
+As it builds on the foundations set by many popular and widely-used languages such as C, C++, Java and Python, much of Go's syntax draws from existing implementations and will be familiar to programmers looking to learn an additional language. However, Go also diverges explicitly from these other languages; listed below are some aspects of Go that may be unfamiliar to learners.
+
+### Arrays and slices
+Arrays are not often seen in Go code. This is because the size of a Go array must be declared at its creation, which limits how flexible an array can be.
+
+To create dynamically-sized arrays, Go introduces the concept of a slice. Slices are not arrays; rather, they are data structures that describe a piece of a separately-stored array. Slices are the topic of several Go blog entries; you can read more about [the internal layout of slices](https://blog.golang.org/go-slices-usage-and-internals) and [the mechanics of using slices](https://blog.golang.org/slices). Exercises on how to use slices are available in [A Tour of Go](https://tour.golang.org/moretypes/7).
+
+### `defer`, `panic` and `recover`
+In addition to traditional control flow mechanisms such as `if`, `for` and `switch`, Go also provides three other ways to alter the flow of a program: `defer`, `panic`, `recover`. These mechanisms can be conceived in terms of exception handling in Java and C++, but also have more general purpose uses.
```

Comment:
I think this section needs to be a lot more precise and rescoped.

Exception handling is kind of special in Go because you rely on multiple returns, e.g. some function returns `(int, int, String, err)` and we always check `if err != nil` to perform exception handling. `panic` is only used if `err != nil` _and_ we don't know what to do. Tl;dr `panic` and `recover` only describe a subset of exception handling
https://blog.golang.org/error-handling-and-go

You could talk about `defer` here, and move `panic` and `recover` into a separate section dedicated to exception handling. I think it's worth doing so because exception handling is quite different in Go compared to other languages.

You can also include an example of `defer` since it's not a common language feature (I think).